pallets/avn-anchor/src/lib.rs

```diff
@@ -175,21 +185,21 @@ pub mod pallet {
                 Error::<T>::HandlerAlreadyRegistered
             );
 
-            ChainHandlers::<T>::try_mutate(&old_handler, |maybe_chain_data| -> DispatchResult {
-                let chain_data = maybe_chain_data.take().ok_or(Error::<T>::ChainNotRegistered)?;
-                ChainHandlers::<T>::insert(&new_handler, chain_data.clone());
+            ChainHandlers::<T>::try_mutate(&old_handler, |maybe_chain_id| -> DispatchResult {
+                let chain_id = maybe_chain_id.take().ok_or(Error::<T>::ChainNotRegistered)?;
+                ChainHandlers::<T>::insert(&new_handler, chain_id);
```

Comment:
Shouldn't we be using `do_update_chain_handler()`? But the same comment applies there too.